code/scripting/hook_conditions.cpp

```diff
@@ -305,5 +306,10 @@ HOOK_CONDITIONS_START(KeyPressConditions)
 	HOOK_CONDITION(KeyPressConditions, "Raw KeyPress", "The key that is pressed, with no consideration for any modifier keys.", keycode, conditionParseRawControl, conditionCompareRawControl);
 HOOK_CONDITIONS_END
 
+HOOK_CONDITIONS_START(CommOrderConditions)
+	HOOK_CONDITION_SHIPP(CommOrderConditions, "", "that sent the order.", source);
+	HOOK_CONDITION(CommOrderConditions, "Object type", "Specifies the type of object that is the target of the order.", target, conditionParseObjectType, conditionIsObjecttype);
+HOOK_CONDITIONS_END
```

Comment:
By adding a `HOOK_CONDITION_SHIP_OBJP` (and similarly doing the prefix thing you did for `HOOK_CONDITION_SHIPP`, you can add a one-liner to allow conditions based on "Target Ship", and "Target Ship Class" and "Target Ship type". These will automatically never match if your objp is not a ship, so you don't need any more additional handling.

Receiving ships can wait for now, since the required OSWPT handling methods don't exist yet, and coding them in here is a bit beyond scope.

Reply:
Done.